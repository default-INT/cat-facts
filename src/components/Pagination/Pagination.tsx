import { memo } from 'react';
import { useRouter } from 'next/router';
import Link from 'next/link';
import { makeLink } from 'utils/makeLink';

interface IProps {
  pageCount: number;
}

export const Pagination = memo(({ pageCount }: IProps) => {
  const { query, pathname } = useRouter();
  const pageNum = Number(query.page);

  if (Number.isNaN(pageNum)) return null;

  return (
    <div className='py-3 mt-10 flex justify-between'>
      {pageNum > 1 && (
        <Link
          className='border-solid p-1 px-5 rounded-2xl border-2 border-opacity-50 border-indigo-600'
          href={makeLink(pathname, { params: { page: pageNum - 1 } })}
        >
          {'<'}
        </Link>
      )}
      {pageNum !== pageCount && (
        <Link
          className='border-solid p-1 px-5 rounded-2xl border-2 border-opacity-50 border-indigo-600'
          href={makeLink(pathname, { params: { page: pageNum + 1 } })}
        >
          {'>'}
        </Link>
      )}
    </div>
  );
});

Pagination.displayName = 'Pagination';
