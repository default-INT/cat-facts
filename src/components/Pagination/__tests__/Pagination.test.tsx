import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import { Pagination } from 'components/Pagination';

jest.mock('next/router', () => ({
  useRouter: () => ({
    push: jest.fn(),
    pathname: '/facts/[page]',
    query: {
      page: 5,
    },
  }),
}));

describe('tests for Pagination', () => {
  it('should match snapshot', () => {
    const { container } = render(<Pagination pageCount={20} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should check href', () => {
    const { container } = render(<Pagination pageCount={20} />);
    const [goPrevPage, goNextPage] = Array.from(container.querySelectorAll('a'));

    expect(goPrevPage).toBeInTheDocument();
    expect(goPrevPage).toHaveAttribute('href', '/facts/4');
    expect(goNextPage).toBeInTheDocument();
    expect(goNextPage).toHaveAttribute('href', '/facts/6');
  });

  it('should render only left arrow', () => {
    const { container } = render(<Pagination pageCount={5} />);
    const arrows = Array.from(container.querySelectorAll('a'));

    expect(arrows.length).toEqual(1);
  });
});
