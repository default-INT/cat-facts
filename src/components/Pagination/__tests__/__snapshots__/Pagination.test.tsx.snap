// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for Pagination should match snapshot 1`] = `
<div
  class="py-3 mt-10 flex justify-between"
>
  <a
    class="border-solid p-1 px-5 rounded-2xl border-2 border-opacity-50 border-indigo-600"
    href="/facts/4"
  >
    &lt;
  </a>
  <a
    class="border-solid p-1 px-5 rounded-2xl border-2 border-opacity-50 border-indigo-600"
    href="/facts/6"
  >
    &gt;
  </a>
</div>
`;
